fulp_modules/bloodsuckers/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,746 @@
+
+/atom/movable/screen/bloodsucker
+	icon = 'fulp_modules/bloodsuckers/icons/actions_bloodsucker.dmi'
+	invisibility = INVISIBILITY_ABSTRACT
+
+/atom/movable/screen/bloodsucker/proc/update_counter()
+	invisibility = 0
+
+/atom/movable/screen/bloodsucker/blood_counter
+	name = "Blood Consumed"
+	icon_state = "blood_display"
+	screen_loc = ui_blood_display
+
+/atom/movable/screen/bloodsucker/rank_counter
+	name = "Bloodsucker Rank"
+	icon_state = "rank"
+	screen_loc = ui_vamprank_display
+
+/atom/movable/screen/bloodsucker/sunlight_counter
+	name = "Solar Flare Timer"
+	icon_state = "sunlight_night"
+	screen_loc = ui_sunlight_display
+
+/mob/living/carbon/proc/handle_bloodsucker()
+	return
+/* -- Just some testing I've done
+	var/datum/antagonist/bloodsucker/bloodsucker = mind.has_antag_datum(/datum/antagonist/bloodsucker)
+	if(mind && hud_used?.blood_display)
+		if(bloodsucker)
+			hud_used.blood_display.invisibility = 0
+			// Placeholders until I can get it to work
+			hud_used.blood_display.maptext = MAPTEXT("<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='#FFDDDD'>'>[round(1)]</font></div>")
+//			hud_used.blood_display.maptext = MAPTEXT("<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='[valuecolor]'>[round(value,1)]</font></div>")
+		else
+			hud_used.blood_display.invisibility = INVISIBILITY_ABSTRACT
+	if(mind && hud_used?.vamprank_display)
+		if(bloodsucker)
+			hud_used.vamprank_display.invisibility = 0
+			// Placeholders until I can get it to work
+			hud_used.vamprank_display.maptext = MAPTEXT("<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='#FFDDDD'>[round(1)]</font></div>")
+//			hud_used.vamprank_display.maptext = MAPTEXT("<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='[valuecolor]'>[round(value,1)]</font></div>")
+		else
+			hud_used.vamprank_display.invisibility = INVISIBILITY_ABSTRACT
+	if(mind && hud_used?.sunlight_display)
+		if(bloodsucker)
+			hud_used.sunlight_display.invisibility = 0
+			// Placeholders until I can get it to work
+			hud_used.sunlight_display.maptext = MAPTEXT("<div align='center' valign='bottom' style='position:relative; top:0px; left:6px'><font color='#FF5555'>[round(1)]</font></div>")
+//			hud_used.sunlight_display.maptext = MAPTEXT("<div align='center' valign='bottom' style='position:relative; top:0px; left:6px'><font color='[valuecolor]'>[value]</font></div>")
+		else
+			hud_used.sunlight_display.invisibility = INVISIBILITY_ABSTRACT
+*/
+
+/datum/antagonist/bloodsucker/proc/update_hud(updateRank=FALSE)
+	// Update Blood Counter
+	if(owner.current.hud_used && owner.current.hud_used.blood_display)
+		if(owner.current.blood_volume > BLOOD_VOLUME_SAFE)
+			valuecolor =  "#FFDDDD"
+		else if(owner.current.blood_volume > BLOOD_VOLUME_BAD)
+			valuecolor =  "#FFAAAA"
```

Comment:
presumably you'd have an else statement after this for if their blood volume is <= BLOOD_VOLUME_BAD, no?